OpenRA.Game/FileSystem/MixFile.cs

```diff
@@ -122,8 +122,14 @@ static MemoryStream Decrypt(uint[] h, Blowfish fish)
 
 		static uint[] ReadBlocks(Stream s, long offset, int count)
 		{
+			if (offset > s.Length)
+				throw new ArgumentOutOfRangeException("offset", "Offset {0} greater than stream length {1}.".F(offset, s.Length));
+
 			s.Seek(offset, SeekOrigin.Begin);
 
+			if (count < 0)
```

Comment:
So you don't change the stream position until you've validated all the arguments.